pipeline/src/utils/calculate_processes.jl

```diff
@@ -0,0 +1,67 @@
+"""
+Internal function for calculating the log of the infections with an informative
+    error message if the infections are not positive definite.
+"""
+function _calc_log_infections(I_t)
+    @assert all(I_t .> 0) "Infections must be positive definite."
+    log.(I_t)
+end
+
+"""
+Internal function for calculating the exponential growth rate with an informative
+    error message if the infections are not positive definite.
+"""
+function _calc_rt(I_t, I0)
+    @assert all(I_t .> 0) "Infections must be positive definite."
+    @assert I0>0 "Initial infections must be positive definite."
+    log.([I0; I_t]) .- log(I0) |> diff
+end
+
+"""
+Internal function for calculating the _instantaneous_ reproduction number `Rt`
+using the method of [Fraser (2007)](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0000758).
+Left truncation in `I_t` is handled by extending the series with backwards exponential
+growth from the initial infections `I0` and the exponential growth rate `init_rt`.
+
+# Arguments
+- `I_t`: Incident infections.
+- `I0`: Initial infections at time zero.
+- `init_rt`: Initial exponential growth rate.
+- `data::EpiData`: An instance of the `EpiData` type containing generation interval data.
+"""
+function _calc_Rt(I_t, I0, init_rt, data::EpiData)
+    @assert all(I_t .> 0) "Log infections must be positive definite."
+    @assert I0>0 "Initial infections must be positive definite."
+    n = length(data.gen_int)
+    rev_pmf = reverse(data.gen_int)
+    aug_I_t = [I0 * exp(-init_rt * (n - i)) for i in 1:n] |>
```

Comment:
I don't love encoding this seeding assumption everywhere (i.e not in `EpiAware` for sure and potentially not in a version of this pipeline that is meant to be more generic). 

I wonder if it would be better to just only return Rt from the start of when there is enough infection data? 

Design wise I also think we should try and have a single I_t vector as the input vs expecting people to treat I_t and I_0 differently? 

The problem with my suggestion is that you then get a different length Rt to everything else but perhaps the solution there is to pad with missingness?

This is all just a thought and happy to move discussion into a new issue

Reply:
> The problem with my suggestion is that you then get a different length Rt to everything else but perhaps the solution there is to pad with missingness?

Yes, this was my reason for going this way (but not changing the underlying `EpiAware` function with a strong decision).